File: servant/src/CardsProject/Content/ArticleCommentService.hs
{-# LANGUAGE OverloadedStrings #-}
module CardsProject.Content.ArticleCommentService
  ( validateArticleComment, hide, unhide, is_reply
  ) where

import CardsProject.Content.Types
import Control.Exception (throwIO)
import System.IO.Error (userError)
import Data.Text (Text)
import Database.SQLite.Simple
import Database.SQLite.Simple.FromField ()
import CardsProject.Db (withDb)

-- Domain service for ArticleComment
validateArticleComment :: NewArticleComment -> Either String NewArticleComment
validateArticleComment body = Right body

-- @invoke behavior stub (no-op)
hide :: Int -> IO ()
hide _eid = throwIO (userError "hide not implemented")

-- @invoke behavior stub (no-op)
unhide :: Int -> IO ()
unhide _eid = throwIO (userError "unhide not implemented")

-- @invoke behavior stub (no-op)
is_reply :: Int -> IO Bool
is_reply _eid = throwIO (userError "is_reply not implemented")

File: laravel/app/Services/Content/ArticleCommentService.php
<?php

namespace App\Services\Content;

use App\Models\Content\ArticleComment;

class ArticleCommentService
{
    public function create(array $data): ArticleComment
    {
        throw new \LogicException('Not implemented');
    }

    public function update(ArticleComment $articleComment, array $data): ArticleComment
    {
        throw new \LogicException('Not implemented');
    }
    public function hide(int $id): void
    {
        $articleComment = ArticleComment::findOrFail($id);
        $articleComment->hide();
        $articleComment->save();
    }

    public function unhide(int $id): void
    {
        $articleComment = ArticleComment::findOrFail($id);
        $articleComment->unhide();
        $articleComment->save();
    }

    public function isReply(int $id): bool
    {
        $articleComment = ArticleComment::findOrFail($id);
        $result = $articleComment->isReply();
        $articleComment->save();
        return $result;
    }
}
